Address most of Marie-Jose's comments

brunorijsman opened the pull request.

BrunoRijsman-AliroQuantum committed:
Address most of Marie-Jose's comments

BrunoRijsman-AliroQuantum committed:
Improve figure

BrunoRijsman-AliroQuantum committed:
Regenerate .txt from .xml

Wojtek242 commented:
Thanks a lot @brunorijsman! I haven't looked at the content changes yet, but one of your commits (I believe the second one) completely reformatted the indentation of the entire xml file which I'd rather not do as my editor will probably insist on undoing all of that. Would you be able to revert the indentation changes for this PR?

brunorijsman commented:
Urghh… I will undo that.  I thought I had turned off Visual Studio Code’s “auto format on save” but evidently not.


> On Jan 13, 2022, at 4:11 PM, Wojciech Kozlowski ***@***.***> wrote:
> 
> 
> Thanks a lot @brunorijsman <https://github.com/brunorijsman>! I haven't looked at the content changes yet, but one of your commits (I believe the second one) completely reformatted the indentation of the entire xml file which I'd rather not do as my editor will probably insist on undoing all of that. Would you be able to revert the indentation changes for this PR?
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/Wojtek242/draft-irtf-qirg-principles/pull/10#issuecomment-1012556603>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAQL5DVRHKC656AKAGJ4ZTUV5EXTANCNFSM5L2LL36A>.
> You are receiving this because you were mentioned.
>

BrunoRijsman-AliroQuantum committed:
Revert "Improve figure"

This reverts commit 7f62fc2335312a64cabea7fbee0989424c840fa6.

BrunoRijsman-AliroQuantum committed:
Redo improve figure

brunorijsman commented:
Fixed it. 

> On Jan 13, 2022, at 4:12 PM, Bruno Rijsman ***@***.***> wrote:
> 
> Urghh… I will undo that.  I thought I had turned off Visual Studio Code’s “auto format on save” but evidently not.
> 
> 
>> On Jan 13, 2022, at 4:11 PM, Wojciech Kozlowski ***@***.*** ***@***.***>> wrote:
>> 
>> 
>> Thanks a lot @brunorijsman <https://github.com/brunorijsman>! I haven't looked at the content changes yet, but one of your commits (I believe the second one) completely reformatted the indentation of the entire xml file which I'd rather not do as my editor will probably insist on undoing all of that. Would you be able to revert the indentation changes for this PR?
>> 
>> —
>> Reply to this email directly, view it on GitHub <https://github.com/Wojtek242/draft-irtf-qirg-principles/pull/10#issuecomment-1012556603>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAQL5DVRHKC656AKAGJ4ZTUV5EXTANCNFSM5L2LL36A>.
>> You are receiving this because you were mentioned.
>> 
>

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -217,13 +217,14 @@
 
    <abstract>
-     <t>The vision of a quantum internet is to fundamentally enhance Internet
+     <t>The vision of a quantum Internet is to enhance existing Internet
```
I would not capitalize "internet" here since it's "**a** quantum internet"

brunorijsman replied:
Done: changed Internet to internet (lower case i)

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -217,13 +217,14 @@
-     imminent, but there is no practical proposal for how to organise, utilise,
-     and manage such networks. In this memo, we attempt to lay down the
+     entanglement. The first quantum entanglement networks have been realised
+     <xref target="Pompili21" />, but there is no practical proposal for how to
```
I'm wondering whether this should also be included: https://www.science.org/doi/10.1126/sciadv.aba0959. It's also entanglement based, but without quantum memories or swapping.

Wojtek242 replied:
No need to act on this, leaving this for the QIRG mailing list.

brunorijsman replied:
Interesting paper; thanks for pointing to it. I can see how the approach described in this paper allows for a rich topology without using memories or swapping, but I do not see how it overcomes the distance limitations of existing QKD (~ 120 km for BB84, ~ 600 km for TF).  The "long distances" referred to in this paper (start of page 4) are only up to 12.6 km.  Anyway... since you said "no need to act on this, leaving it to the mailing list" I am marking the comment as resolved.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -245,9 +246,11 @@
-     otherwise impossible to realise <xref target="VanMeterBook" />.</t>
+     otherwise impossible to realise <xref target="VanMeterBook" />,
+     such as clustered or distributed quantum computing, quantum 
+     secure communications, and distributed quantum sensing.</t>
```
I'd also add secure delegated quantum computing (i.e. blind quantum computing).

Wojtek242 replied:
Should also add references to these as well. If you feel like it, you can have a look at the intro in https://dl.acm.org/doi/pdf/10.1145/3386367.3431293 and add them. Feel free, to leave this alone as well and I'll add them later.

brunorijsman replied:
Added "blind quantum computing" and "clock synchronization". I will leave it to you to add the references.

Wojtek242 replied:
Thanks, I'll add them after merging your PR

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -260,23 +263,25 @@
-     quantum information are now imminent <xref target="Castelvecchi18" /> <xref
-     target="Wehner18" />. Whilst a lot of effort has gone into physically
+     <t>The first fully quantum networks capable of transmitting and managing
+     entangled quantum states (defined in <xref target="entanglement" />) in order
```
I think the better solution is to just avoid mentioning entanglement or quantum states until they are defined.

I propose to rephrase this sentence as follows: "The first fully quantum networks capable of transmitting and manipulating distributed quantum information ("prepare and measure" stage in [WEHNER18]) via an intermediate quantum node have recently been realized"

brunorijsman replied:
Did you notice that I added a reference to [Pompili21] (https://arxiv.org/pdf/2102.04471.pdf).  It seems to me that this goes beyond the "prepare and measure" stage.

Wojtek242 replied:
Yes, I've seen. But I was just trying to define "fully quantum" network without mentioning entangled states and I think that defining it as "prepare and measure" stage (which defines minimum end-node capabilities) via an intermediate node (so that it's not a p2p link and actually some sort of network) is suitable. And yes, the Delft network is quantum memory stage anyway, but I don't think we have to draw the line that high for the purposes of this draft.

brunorijsman replied:
I took the easy way out: mention neither entanglement nor prepare-and-measurement at this early stage in the document:

"The first fully quantum networks capable of sending, receiving, and manipulating distributed quantum information have recently been realized [Pompili21]."

Pushed another commit with this change.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -260,23 +263,25 @@
+     entangled quantum states (defined in <xref target="entanglement" />) in order
+     to send, receive, and manipulate distributed quantum information have 
+     recently been realized <xref target="Pompili21" />
+     <xref target="Castelvecchi18" /> <xref target="Wehner18" />. 
```
The Castelvecchi and Wehner references need to be moved or removed. I'd say remove both.

brunorijsman replied:
I have removed the references to Castelvecchi and Wehner.  Note that this contradicts your previous comment where you suggested changing the sentence to "The first fully quantum networks capable of transmitting and manipulating distributed quantum information ("prepare and measure" stage in [WEHNER18]) via an intermediate quantum node have recently been realized" which *does* include the Wehner reference.

Wojtek242 replied:
Haha yes. I meant kick out the ones at the end that were placed there originally as a reference relative to "imminently realized" to being relative to the mention of quantum network stages.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -260,23 +263,25 @@
      physically connect our devices and send data, but all sending, receiving,
      buffer management, connection synchronisation, and so on, must be managed
-     by the application itself at a level below conventional assembly language,
+     by the application itself rather than being managed by the network stack.
```
I don't think this rephrasing reflects the original meaning. I think I'd rephrase to "by the application itself using low-level instructions sets, including quantum assembly (QASM) languages, rather than being managed by a network stack".

brunorijsman replied:
Restored the language that you suggested. All added a reference to the OpenQASM 3.0 paper.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -308,7 +313,8 @@
 
-       <t>where |X> is Dirac's ket notation for a quantum state, here the
+       <t>where |X> is Dirac's ket notation for a quantum state (the value that a 
+       qubit holds), here the
```
Quantum state is more than that, but it's a good point, there probably needs to be some text that explains what a quantum state is and how it is different and relates to a qubit.

Wojtek242 replied:
On my TODO list

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -534,14 +546,16 @@
-         state we have tried to create. It expresses the probability that one
-         state will pass a test to identify as the other. Fidelity is an
+         state we have tried to create. It expresses the probability measuring
+         the actual state yields the same result as measuring the desired target 
```
Perhaps, "It expresses the probability that the state will behave exactly the same as our desired state"? Yielding the same result is not necessarily correct I think as that will depend on the measurement.

brunorijsman replied:
Don't you think this may be splitting the hair too finely for a classical network engineer and making the explanation less clear rather than more clear the process?  I thought it was rather obvious from the context that we are doing the same measurement on the desired state and the target state... Anyway, if you feel strongly about it, I can change the language to what you suggested.

Wojtek242 replied:
I think that at the same time it is important to be correct though. I do think that some variation "It expresses the probability that the state will behave exactly the same as our desired state" or "It expresses the probability that the state will **mimic the behaviour of the** desired state" is a good balance between the original phrase (which was paraphrased from wikipedia) and your suggestion which is not correct in all circumstances.

brunorijsman replied:
I changed the description of fidelity to the first suggestion that you provided.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -555,7 +569,8 @@
          sacrifice rate for higher fidelity) or by employing quantum error
-         detection and correction mechanisms.</t>
+         detection and correction mechanisms (see <xref target="VanMeterBook" />
+         chapter 11).</t>
```
Should probably also add a reference to the generations paper "muhlaridan" I think...

brunorijsman replied:
Happy to add it. However, Googling "muhlaridan generations" (or variations thereof) did not yield anything.  Do you have a link to the paper?

Wojtek242 replied:
It's already in this draft as "Mural16". Sorry, I just spelled it terribly bad.

brunorijsman replied:
Added reference.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -573,7 +588,9 @@
-         quantum networks.</t>
+         quantum networks. Until then, quantum networks rely on entanglement
+         swapping (<xref target="es" />) and teleportation
+         (<xref target="teleportation" />).</t>
```
Can make a reference to the generations paper again and also refer to the section where the generations are explained.

brunorijsman replied:
Ditto - Do you have a link for the paper that you have in mind? I can add it (and the reference to the section).

brunorijsman replied:
Added ref to Mural16 fig 2 here.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -794,7 +811,7 @@
          long-distance entangled pairs via entanglement swapping are called
          quantum repeaters in academic literature <xref target="Briegel98" />
-         and we will use the same terminology in this memo.</t>
+         and we will use the same terminology in this draft.</t>
```
Is this normal/preferred? It won't be a draft after it's published, no?

brunorijsman replied:
I'm with you, Woj. But Marie-Jose's response was firm on this. Leaving "draft".

Wojtek242 replied:
Ack

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -1009,7 +1026,9 @@
            makes control algorithms and optimisation of quantum networks
-           different from classical ones. Note that third generation quantum
+           different from classical ones, in the sense that they are more akin
+           to a stateful distributed computation than stateless packet-by-packet
```
I'd just say they are necessarily "stateful" without calling it "stateful distributed computation. Whilst I know some like this phrasing, I know people who will disagree with it (and I'm neutral on this). The document is meant to be a consensus so I'd avoid it.

brunorijsman replied:
Updated according to your suggestion.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -1275,27 +1296,27 @@ CC - classical channel (can consist of many classical links)
-         once the pair is generated. On the other hand, nitrogen-vacancy
-         architectures <xref target="Hensen15" /> offer a much greater degree
+         once the pair is generated. On the other hand, nitrogen-vacancy based
+         hardware <xref target="Hensen15" /> offers a much greater degree
```
We can also add "trapped ions [REF: https://www.nature.com/articles/nature06118]" reference for better coverage.

brunorijsman replied:
No objection to adding the reference, but what do you suggest we say single-atom qubits to contrast them with atomic ensembles and NV-center?

Wojtek242 replied:
Single-atom qubits is too broad as that could also be neutral atoms.

brunorijsman replied:
Yeah, sorry typo. I meant to say "ion-trap" instead of"single-atom". Anyway, can you suggest some specific language to add?

Wojtek242 replied:
On the other hand, nitrogen-vacancy <xref target="Hensen15" /> or trapped ion <NEW REF> based hardware offer a much greater degree...

Wojtek242 replied:
I wouldn't bother contrasting, just adding to the list so that it's not so centered around NVs

brunorijsman replied:
Added mention of ion trap, with reference to Moehring07, without specific pros and cons.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -1369,6 +1390,10 @@ CC - classical channel (can consist of many classical links)
          build up network-wide connectivity.</t>
 
+         <t>This rate is expected to improve over time as quantum network
+         technology evolves. Rates much higher than 10 Hz are needed to enable
```
10 Hz seems a bit random without a reference.

brunorijsman replied:
I copied the 10 Hz from the paragraph right above it, which was already there and which does have a reference. I double checked, and the reference actually has the number 10Hz in it (figures 17 through 22 and table 3 all have throughputs peaking below 10 Hz)

Wojtek242 replied:
Ah yes, that makes sense. Silly GitHub hiding the paragraph above.

Wojtek242 commented on draft-irtf-qirg-principles-09-rc.xml:
```diff
@@ -1602,7 +1635,13 @@ CC - classical channel (can consist of many classical links)
+           quantum secure networks require the distribution of quantum state;
+           the distribution of classical state alone is not sufficient.
+           Currently, the only feasible way of reliably distributing quantum state
+           is using teleportation, which requires distribution of entanglement.
```
That's not strictly correct. Teleportation can indeed do it, but if the state you need can be constructed from Bell states then you don't need teleportation. I would rephrase this a bit.

I would state that "...quantum applications consume entanglement like a resource..." and then state that this fundamental resource can be either used directly in an application, e.g. QKD which simply measures the entangled qubits to obtain a secret key, used to build more complex abstraction and operations, e.g. distributed CNOT gates (https://www.quantum-network.com/applications/distributed-cnot/),  or to directly teleport quantum information.

brunorijsman replied:
Updated the sentence as you suggested.  Also added the reference to the QNE distributed CNOT page (as well as another reference to the QNE teleportation page).

Wojtek242 commented:
See also e-mail on QIRG mailing list: https://mailarchive.ietf.org/arch/msg/qirg/TNPvzu90ImcvvCXhT__clK6xbSc/

BrunoRijsman-AliroQuantum committed:
Address some of Woj's review comments on the pull request

BrunoRijsman-AliroQuantum committed:
Avoid mentioning entanglement or prepare-and-measure early in the draft.

BrunoRijsman-AliroQuantum committed:
Restore comment about using low-level QASM-like languages

BrunoRijsman-AliroQuantum committed:
More accurate description of fidelity.

BrunoRijsman-AliroQuantum committed:
Add Mural16 reference

BrunoRijsman-AliroQuantum committed:
Add 2nd ref to Mural16

BrunoRijsman-AliroQuantum committed:
Also mention ion-trap hardware with reference to  Moehring07

brunorijsman commented:
Okay, I think I have resolved all comments. We still need to run xml2rfc to generate the new text.

> On Jan 21, 2022, at 11:15 AM, Wojciech Kozlowski ***@***.***> wrote:
> 
> 
> @Wojtek242 commented on this pull request.
> 
> In draft-irtf-qirg-principles-09-rc.xml <https://github.com/Wojtek242/draft-irtf-qirg-principles/pull/10#discussion_r789852599>:
> 
> > @@ -308,7 +313,8 @@
>  
>         <t>a |0> + b |1>,</t>
>  
> -       <t>where |X> is Dirac's ket notation for a quantum state, here the
> +       <t>where |X> is Dirac's ket notation for a quantum state (the value that a 
> +       qubit holds), here the
> On my TODO list
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/Wojtek242/draft-irtf-qirg-principles/pull/10#discussion_r789852599>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAQL5BZM62Q5B3KJBZNCYLUXGIDPANCNFSM5L2LL36A>.
> You are receiving this because you were mentioned.
>

Wojtek242 commented:
Awesome, thanks!

Wojtek242 commented:
Can you run xml2rfc to regenerate the text file. Then I'll just do a "squash and merge" and the update will be all in one commit.

BrunoRijsman-AliroQuantum committed:
Run xml2rfc and produce .txt file

Also fix some issues in the XML file that were discovered as a result of running xml2rfc

brunorijsman commented:
I ran xml2rfc, produced the .txt files, and fixed some issues that xml2rfc found in the xml file

Go ahead, squash and merge

> On Jan 24, 2022, at 8:35 AM, Wojciech Kozlowski ***@***.***> wrote:
> 
> 
> Can you run xml2rfc to regenerate the text file. Then I'll just do a "squash and merge" and the update will be all in one commit.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/Wojtek242/draft-irtf-qirg-principles/pull/10#issuecomment-1020164730>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAAQL5FS6N67UX5L4ZTLZFLUXVPSLANCNFSM5L2LL36A>.
> You are receiving this because you were mentioned.
>
